self-core-impl/src/main/java/com/selfxdsd/core/CachingJsonResources.java

```diff
@@ -0,0 +1,241 @@
+ * All rights reserved.
+ * <p>
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"),
+ * to read the Software only. Permission is hereby NOT GRANTED to use, copy,
+ * modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software.
+ * <p>
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ * OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
+ * OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.selfxdsd.core;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.json.Json;
+import javax.json.JsonValue;
+import java.io.StringReader;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+
+/**
+ * Cacheable JSON Resources used by Provider.
+ * @author criske
+ * @version $Id$
+ * @since 0.0.79
+ */
+public final class CachingJsonResources implements JsonResources {
+
+    /**
+     * Logger.
+     */
+    private static final Logger LOG = LoggerFactory.getLogger(
+        CachingJsonResources.class
+    );
+
+
+    /**
+     * JsonResources delegate.
+     */
+    private final JsonResources delegate;
+
+
+    /**
+     * Json storage.
+     */
+    private final JsonStorage jsonStorage;
+
+    /**
+     * Ctor.
+     * @param delegate JsonResources delegate.
+     * @param jsonStorage JSON storage.
+     */
+    public CachingJsonResources(final JsonResources delegate,
+                                final JsonStorage jsonStorage) {
+        this.delegate = delegate;
+        this.jsonStorage = jsonStorage;
+    }
+
+    /**
+     * Ctor.
+     * @param delegate JsonResources delegate.
+     */
+    public CachingJsonResources(final JsonResources delegate) {
+        this(delegate, new JsonStorage.InMemory());
+    }
+
+
+    @Override
+    public JsonResources authenticated(final AccessToken accessToken) {
+        return new CachingJsonResources(
+            this.delegate.authenticated(accessToken),
+            this.jsonStorage
+        );
+    }
+
+    @Override
+    public Resource get(final URI uri) {
+        return this.tryGetFromCache(uri, Collections::emptyMap);
```

Comment:
@criske Can we call the overloaded method instead? ``this.get(uri, () -> new HashMap<>());``. I think it's more straight forward.